app/src/Navigators/Main.tsx

```diff
@@ -1,62 +1,144 @@
 /**
  * @file Defines the main screens that have tabs at the bottom of the app, e.g. Projects, Events.
  */
-
 import React from 'react'
 import { createBottomTabNavigator } from '@react-navigation/bottom-tabs'
 import { ListContainer } from '@/Containers'
-import Entypo from 'react-native-vector-icons/Entypo'
 import { ListType } from '@/Containers/ListContainer'
-import { useTheme } from '@/Hooks'
-import ProfileContainer from '@/NativeBase/Containers/ProfileContainer'
-import VerticalStackContainer from '@/NativeBase/Containers/VerticalStackContainer'
+import {
+  ProfileContainer,
+  VerticalStackContainer,
+} from '@/NativeBase/Containers'
+import { EventTicket, Home, Projects, User } from '@/NativeBase/Assets/Icons'
+import { View, useColorMode, ColorMode } from 'native-base'
+import StaTheme from '@/NativeBase/Theme/StaTheme'
+
+type navBarProps = {
+  focused: boolean
+  colorMode: ColorMode
+}
 
 const Tab = createBottomTabNavigator()
 
-// @refresh reset
+const TabBarIconWrapper = ({ focused, colorMode }: navBarProps) => (
+  <View
+    borderTopColor={
+      focused
+        ? colorMode === 'light'
+          ? StaTheme.colors.primary['100']
+          : StaTheme.colors.primary['40']
+        : 'transparent'
+    }
+    borderTopWidth={2}
+    style={{
+      width: 97,
+    }}
+  />
+)
+
+const iconColor = ({ focused, colorMode }: navBarProps) => {
+  return focused
+    ? colorMode === 'light'
+      ? StaTheme.colors.primary['100']
+      : StaTheme.colors.primary['40']
+    : colorMode === 'light'
+    ? StaTheme.colors.text['100']
+    : StaTheme.colors.textDarkMode['100']
+}
+
+const HomeTabBarIcon = ({ focused, colorMode }: navBarProps) => (
+  <>
+    <TabBarIconWrapper focused={focused} colorMode={colorMode} />
+    <View
+      style={{
+        padding: 4,
```

Comment:
How come the padding varies for each icon -- is it because the icons are different sizes?  If so, there might be another way to approach this, with a layout that centres the icon inside an area?